### ISSUE TITLE
feature request support zip or rar file format

### ISSUE THREAD

**Issue by alexwallis** [litchie/ge-public#72]
Although golden ear is an audio player, I think it would be good if the zip or rar file format could be supported,
this is because if I use an app such as dropbox to download content to my phone you can't download entire folders, you have to use the open in function on individual files, however if golden ear supported zip or rar you could use the open in function to export the file to golden ear then have it extract the contents. At the moment you can only open individual flac files in golden ear. Of course proper cloud support would also be good as outlined in issue #21 as downloading content from the cloud isn't so easy at the moment.

**Comment by litchie** [comment]
Thanks. You're not the only one asking for this feature. We'll add it in next update.

**Comment by alexwallis** [comment]
The unzip and unrar features seem to be working,
however I have noticed a few things, firstly whenever you extract anything the extraction path isn't correct, you first get a folder named after the original rar or zip file, and then inside that you get another folder named after the rar or zip file and then finally inside that is the extracted contents.
I would like it so that you get one folder created during the extraction and the extraction mirrors the path of the rar file itself.
Another idea I had is a setting to automatically delete the rar or zip file on successful extraction.
During extraction would it be possible to show a progress bar that indicates the extraction progress? as at the moment all VoiceOver says is in progress during extraction.

I hope soon we will see support for cloud services to import files.

**Comment by litchie** [comment]
Agreed. We should be smarter in deciding whether to create a new level of folder or not.

Do you want the progress announced every 10 percent?

**Comment by alexwallis** [comment]
That's a very difficult decision as with large zip files that would be fine, but with smaller files could a situation happen where VoiceOver starts reading the extraction progress but the extraction finishes while VoiceOver is still announcing the percentages? Or would it be possible for you to iinterrupt the percentage reading as soon as the extraction is successfully completed even if VoiceOver hasn't finished reading them.

**Comment by litchie** [comment]
we can figure out a way. what's other app doing in similar situation?

**Comment by alexwallis** [comment]
I am not sure I haven't really encountered apps with this type of situation, I had originally imagined manually checking the progress bar, but if this could be read automatically as long as VoiceOver wasn't reading out of date information I think automatic reading would be better. Maybe every %10 might be two much, perhaps every %25 might be better for automatic reading, with the option of manually checking the progress bar in between automatic readings.
